packages/abi/src/parser/specifications/v1/map-attribute.ts

```diff
@@ -0,0 +1,34 @@
+import { assertUnreachable } from '@fuel-ts/utils';
+
+import type { AbiFunctionAttribute } from '../../abi';
+
+import type { AbiFunctionAttributeV1 } from './specification';
+
+export const mapAttribute = (attribute: AbiFunctionAttributeV1): AbiFunctionAttribute => {
```

Comment:
Given the `storage` and `doc-comment` types are identical and the `Inline` argument param is either `never` or `always` this could be simplified to 

```suggestion
export const mapAttribute = (attribute: AbiFunctionAttributeV1): AbiFunctionAttribute => {
  const { name, arguments: args } = attribute;
  
  switch (name) {
    case 'inline':
      return { name, arguments: args[0] };
    case 'storage':
    case 'doc-comment':
      return { name, arguments: args };
    case 'payable':
    case 'test':
      return { name };
    default:
      return assertUnreachable(attribute);
  }
};
```

Reply:
This doesn't work due to the mismatch of types between `storage` and `doc-comment`:

```console
Type '{ name: "storage" | "doc-comment"; arguments: readonly ("read" | "write")[] | string[]; }' is not assignable to type 'AbiFunctionAttribute'.
  Types of property 'name' are incompatible.
    Type '"storage" | "doc-comment"' is not assignable to type '"doc-comment"'.
      Type '"storage"' is not assignable to type '"doc-comment"'.ts(2322)
```

Reply:
Right, my bad. This could work though

```ts

export const mapAttribute = (attribute: AbiFunctionAttributeV1): AbiFunctionAttribute => {
  const { name, arguments: args } = attribute;

  switch (name) {
    case 'inline':
      return { name, arguments: args[0] };
    case 'storage':
      return { name: 'storage', arguments: args };
    case 'doc-comment':
      return { name, arguments: args };
    case 'payable':
    case 'test':
      return { name };
    default:
      return assertUnreachable(attribute);
  }
};
``` 

Which I'd still say is a bit cleaner.

Reply:
b2d6bdc7b2fcce0669bb22f7692ee8e989dc9266